src/main/java/de/dennisguse/opentracks/io/file/importer/AbstractFileTrackImporter.java

```diff
@@ -126,7 +127,7 @@ public void characters(char[] ch, int start, int length) {
     }
 
     @Override
-    public Track.Id importFile(InputStream inputStream) {
+    public Track.Id importFile(InputStream inputStream)  throws SAXException, SQLiteConstraintException, IOException, ParserConfigurationException {
```

Comment:
Offtopic: have I already expressed my dislike of Javas checked Exceptions? 

This long list could be omitted by wrapping it in a RuntimeException at line 144, but it is a matter of taste.

Reply:
I don't like either.

Do you mean something like this (in the line 144)?
```java
throw new RuntimeException(e);
```

Then, in ImportService, Can I know what exception was triggered? I need it so I know if the error was an already exists (SQLiteConstraintException) or a parser error (SAXException or ParserConfigurationException), so I can inform to the final user in ImportActivity.

Reply:
That I didn't check, so you need to know the type of exception where you catch it?
You can call getCause() to get the wrapped exception, but maybe this is not the ideal thing to do here.

Reply:
Yes, I need the type of exception to show to the user the cause of the errors in the ImportActivity (look the third screenshot). I do that because I think is really important the user knows the cause of the error (silent errors are unuseful ones). This way the user knows if it is OpenTracks's fault or its files are badly parsed, for example.

Reply:
A solution in the middle could be to only distinguish between the SQLiteConstraintException and the others. The first is because of duplicate UUID and the others represent errors reading the file. 
But then you would need two catch clauses, 1 for SQLiteConstraintException which is just rethrown an one for the others to get wrapped into a RuntimeException. 
I suggest not spending too much time here now. When we refactor the UUID check, this might change as well.

Or dirty: just declare Exception here and don't care. 😉

Reply:
Thank you @pstorch 
I'll look at this.

Reply:
My two cents: don't pass a RuntimeException or even Exception.
I have seen larger projects where in higher layers of the code have hard time to figure out what broke (just by reading the code).
Nowadays, I am big fan of just declaring a custom (so to say "business") exception for each case the implementation needs to distinguish and add the originally catched exception as cause.
Then it is easy to understand which exceptions were actually expected and the internal implementation logic is not exposed.

PS: don't declare exception as it is really hard to understand what potential issues might arise.
PPS: The last months I spent a lot of time refactoring and saw things like this: 
```
try {
 Object a = // call to some internal object
 return a;
} catch (Exception e) {
  return null;
}
```
And we are still working on such things, as it takes quite some time to figure out why it is working at all (and if the catch is actually needed etc.). And many cases the exception originates from somewhere in some library and it was passed to some higher layer and then somebody added a try-catch there.
Not too much fun...

Reply:
I feel your pain from larger codebases. Saw all kinds of strange exception handling.

I would suggest to derive the BusinessException from RuntimeException, so you don't have to declare it everywhere only catch where you know what to do.

Reply:
In my last commit I've added custom exceptions to distinguish the origin of the error's cause.

I don't know if I've understanded you, @dennisguse and @pstorch is that what you talked about to do? It's something I've never done...